controller.rst

```diff
@@ -606,13 +606,16 @@ The ``file()`` helper provides some arguments to configure its behavior::
 Final Thoughts
 --------------
 
-Whenever you create a page, you'll ultimately need to write some code that
-contains the logic for that page. In Symfony, this is called a controller,
-and it's a PHP function where you can do anything in order to return the
-final ``Response`` object that will be returned to the user.
+In Symfony, a controller is usualll a class method which is used to accept requests,
+and return a ``Response`` object. Symfony provides an  ``AbstractController`` which
+can be used to extend the controller class, and alllows access to some frequently used utilties 
+such as ``render`` and ``redirectToRoute``. The ``AbstractController`` also 
+provides the ``createNotFoundException`` utility which is used to return a page
+not found response.
 
-To make life easier, you'll probably extend the base ``AbstractController`` class because
-this gives access to shortcut methods (like ``render()`` and ``redirectToRoute()``).
+Services are accesible in controllers by typehinting them in the arguments of the controller methods.
```

Comment:
I don't like this change ... I think it's too early to talk about services. The previous text, which mentioned but didn't explain services looked better to me.

Reply:
I  have removed the paragraph which describes services.